app/src/main/kotlin/compiler/semantic_analysis/VariablePropertiesAnalyzer.kt

```diff
@@ -77,8 +85,12 @@ object VariablePropertiesAnalyzer {
                 is Function -> {
                     node.parameters.forEach {
                         variableProperties[it] = VariableProperties(node)
-                        // scope of the inner function has not begun yet
-                        it.defaultValue?.let { defValue -> analyzeVariables(defValue, currentFunction) }
+                        if (it.defaultValue != null) {
+                            // scope of the inner function has not begun yet
+                            analyzeVariables(it.defaultValue, currentFunction)
+                            // TODO: uncomment after updating tests
```

Comment:
I will add tests to this after you approve the shape of the changes (that it should be as Rafał noticed in his comment).